mobly/controllers/android_device_lib/services/logcat.py

```diff
@@ -0,0 +1,199 @@
+                    try:
+                        line = f.readline()
+                        if not line:
+                            break
+                    except:
+                        continue
+                    line_time = line[:mobly_logger.log_line_timestamp_len]
+                    if not mobly_logger.is_valid_logline_timestamp(line_time):
+                        continue
+                    if self._is_timestamp_in_range(line_time, begin_time,
+                                                   end_time):
+                        in_range = True
+                        if not line.endswith('\n'):
+                            line += '\n'
+                        out.write(line)
+                    else:
+                        if in_range:
+                            break
+
+    def start(self, configs=None):
+        """Starts a standing adb logcat collection.
+
+        The collection runs in a separate subprocess and saves logs in a file.
+
+        Args:
+            configs: Conifg object.
+        """
+        if self._adb_logcat_process:
+            raise Error(
+                self._ad,
+                'Logcat thread is already running, cannot start another one.')
+        configs = configs if configs else self._configs
+        if configs.clear_log:
+            self.clear_adb_log()
+
+        self._enable_logpersist()
+
+        f_name = 'adblog,%s,%s.txt' % (self._ad.model,
+                                       self._ad._normalized_serial)
+        utils.create_dir(self._ad.log_path)
+        logcat_file_path = os.path.join(self._ad.log_path, f_name)
+        cmd = '"%s" -s %s logcat -v threadtime %s >> "%s"' % (
+            adb.ADB, self._ad.serial, configs.logcat_params, logcat_file_path)
+        process = utils.start_standing_subprocess(cmd, shell=True)
+        self._adb_logcat_process = process
+        self.adb_logcat_file_path = logcat_file_path
+
+    def stop(self):
+        """Stops the adb logcat service."""
+        if not self._adb_logcat_process:
+            return
+        try:
+            utils.stop_standing_subprocess(self._adb_logcat_process)
+        except:
+            self._ad.log.exception('Failed to stop adb logcat.')
+        self._adb_logcat_process = None
+
+    def pause(self):
+        """Pauses logcat for usb disconnect."""
+        self.stop()
+        # Clears cached adb content, so that the next time start_adb_logcat()
+        # won't produce duplicated logs to log file.
+        # This helps disconnection that caused by, e.g., USB off; at the
+        # cost of losing logs at disconnection caused by reboot.
+        self.clear_adb_log()
+
+    def resume(self, configs=None):
+        """Resumes a paused logcat service.
+
+        Args:
+            configs: Not used.
+        """
+        # Do not clear device log at this time. Otherwise the log during USB
+        # disconnection will be lost.
+        configs = logcat.Config(clear_log=False)
```

Comment:
You have `configs=None` in the argument list but then you assign `configs` to `logcat.Config(clear_log=False)` immediately after.

Reply:
I don't know what you mean... can you clarify?

Reply:
Oh I see.
Because for `logcat`, the resume should always use `clear_log=False`
Come to think of it, maybe `resume` should not take an arg at all?